tests/integration/agents/k8s/test_k8s_agent.py

```diff
@@ -0,0 +1,314 @@
+from unittest.mock import Mock
+
+import pytest
+from deepeval.metrics import (
+    FaithfulnessMetric,
+    GEval,
+)
+from deepeval.test_case import LLMTestCaseParams
+from langchain_core.messages import AIMessage, HumanMessage, SystemMessage
+
+from agents.common.state import SubTask
+from agents.k8s.agent import KubernetesAgent
+from agents.k8s.state import KubernetesAgentState
+from services.k8s import IK8sClient
+from utils.models.factory import ModelType
+from utils.settings import DEEPEVAL_TESTCASE_VERBOSE
+
+AGENT_STEPS_NUMBER = 25
+
+
+@pytest.fixture
+def correctness_metric(evaluator_model):
+    return GEval(
+        name="Correctness",
+        evaluation_params=[
+            LLMTestCaseParams.ACTUAL_OUTPUT,
+            LLMTestCaseParams.EXPECTED_OUTPUT,
+        ],
+        evaluation_steps=[
+            "Evaluate whether two answers are semantically similar or convey the same meaning."
+            "Check whether the facts in 'actual output' contradict any facts in 'expected output'",
+            "Lightly penalize omissions of detail, focusing on the main idea",
+            "Vague language are permissible",
+        ],
+        model=evaluator_model,
+        threshold=0.7,
+    )
+
+
+@pytest.fixture
+def faithfulness_metric(evaluator_model):
+    return FaithfulnessMetric(
+        threshold=0.7,
+        model=evaluator_model,
+        include_reason=True,
+        verbose_mode=DEEPEVAL_TESTCASE_VERBOSE,
+    )
+
+
+@pytest.fixture
+def mock_k8s_client():
+    return Mock(spec_set=IK8sClient)
+
+
+@pytest.fixture
+def k8s_agent(app_models):
+    return KubernetesAgent(app_models.get(ModelType.GPT4O))
+
+
+@pytest.mark.parametrize(
+    "state,expected_tool_call",
+    [
+        # - Verifies agent makes correct k8s_query_tool call
+        (
+            KubernetesAgentState(
+                agent_messages=[],
+                messages=[
+                    SystemMessage(
+                        content="The user query is related to: {'resource_api_version': 'gateway.kyma-project.io/v1beta1', 'resource_namespace': 'kyma-app-apirule-broken'}"
+                    ),
+                    HumanMessage(content="What is wrong with api rule?"),
+                ],
+                subtasks=[
+                    {
+                        "description": "What is wrong with ApiRule?",
+                        "task_title": "What is wrong with ApiRule?",
+                        "assigned_to": "KubernetesAgent",
+                    }
+                ],
+                my_task=SubTask(
+                    description="What is wrong with api rule?",
+                    task_title="What is wrong with api rule?",
+                    assigned_to="KymaAgent",
+                ),
+                k8s_client=Mock(spec_set=IK8sClient),  # noqa
+                is_last_step=False,
+                remaining_steps=AGENT_STEPS_NUMBER,
+            ),
+            "k8s_query_tool",
+        ),
```

Comment:
please delete this as it is Kyma Agent related query.  ApiRule is Kyma resource.